includes/class-wc-install.php

```diff
@@ -565,7 +569,24 @@ private static function create_tables() {
 
 		// Add constraint to download logs if the columns matches.
 		if ( ! empty( $download_permissions_column_type ) && ! empty( $download_log_column_type ) && $download_permissions_column_type === $download_log_column_type ) {
-			$wpdb->query( "ALTER TABLE {$wpdb->prefix}wc_download_log ADD FOREIGN KEY (permission_id) REFERENCES {$wpdb->prefix}woocommerce_downloadable_product_permissions(permission_id) ON DELETE CASCADE" );
+			$wpdb->query( "
+				IF NOT EXISTS (
+					SELECT NULL FROM information_schema.TABLE_CONSTRAINTS
+					WHERE
+						CONSTRAINT_SCHEMA = '{$wpdb->dbname}' AND
+						(
+							CONSTRAINT_NAME LIKE '%wc_download_log_ib%' OR
```

Comment:
Now that I cleanup routine has been added, I don't think this query should still be checking if foreign keys matching `%wc_download_log_ib%` exists. It should only check if a key named `fk_wc_download_log_permission_id` exists.

If we don't create a `fk_wc_download_log_permission_id` foreign key if a `%wc_download_log_ib%` foreign key exists, sites updating to WC 3.4.3 from any of the previous 3.4.x versions will be left with no foreign keys after `wc_update_343_cleanup_foreign_keys()` runs.